frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMacheteStartupActivity.java

```diff
@@ -0,0 +1,38 @@
+package com.virtuslab.gitmachete.frontend.ui.impl.root;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.startup.StartupActivity;
+import com.intellij.openapi.vcs.VcsNotifier;
+import com.intellij.openapi.wm.ToolWindowId;
+import com.intellij.openapi.wm.ToolWindowManager;
+import lombok.CustomLog;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+@CustomLog
+public class GitMacheteStartupActivity implements StartupActivity {
+
+  @Override
+  @UIEffect
+  public void runActivity(Project project) {
+    addOpenGitMacheteTabListener(project);
+  }
+
+  @UIEffect
+  private void addOpenGitMacheteTabListener(Project project) {
+    var toolWindowManager = ToolWindowManager.getInstance(project);
+    var toolWindow = toolWindowManager.getToolWindow(ToolWindowId.VCS);
+
+    if (toolWindow == null) {
+      LOG.debug("VCS tool window does not exist");
+      VcsNotifier.getInstance(project).notifyWarning(
+          getString("action.GitMachete.OpenMacheteTabAction.notification.title.could-not-open-tab"), // t0d0: proper msg here
+          getString("action.GitMachete.OpenMacheteTabAction.notification.message.no-git")); // t0d0: proper msg here
+      return;
+    }
+
+    var listener = new RediscoverSuggester(project);
```

Comment:
Hmm I run IDE against the project where .git/machete has been modified in July:
```
$ stat .git/machete
  File: .git/machete
  Size: 142       	Blocks: 8          IO Block: 4096   regular file
Device: fd03h/64771d	Inode: 16259565    Links: 1
Access: (0664/-rw-rw-r--)  Uid: ( 1000/pawellipski)   Gid: ( 1000/pawellipski)
Context: unconfined_u:object_r:user_home_t:s0
Access: 2020-10-15 21:49:28.396911004 +0200
Modify: 2020-07-21 17:03:20.211654006 +0200
Change: 2020-07-21 17:03:20.211654006 +0200
 Birth: 2020-07-21 17:03:20.210654002 +0200
```
but nothing really showed up... or is the core functionality not ready yet? :thinking:

Reply:
whoops, strange :thinking: did you try closing and reopening git machete tab :xd:

Reply:
Hmm this isn't even a heisenbug, it just doesn't seem to work (I tested on a few other old projects)... would be nice if you put some logs in the related codebase that I could send you :thinking:

Reply:
I do perform some logging here and there :deciduous_tree: can you retry?

Reply:
Okay! working now

Reply:
Hmmm.... would it be easily possible to show the branch layout before while the rediscover dialog is shown? As for now, when the rediscover dialog is showing, there's nothing in the graph table :thinking:

Reply:
(maybe the dialog can be non-modal?)

Reply:
@mkondratek what's the resolution on the above? (https://github.com/VirtusLab/git-machete-intellij-plugin/pull/621#discussion_r506924768)

Reply:
done